Sources/Tools/plLocalizationEditor/plAddDlgs.cpp

```diff
@@ -161,10 +161,7 @@ std::vector<ST::string> IGetAllLanguageNames()
 
     for (int curLocale = 0; curLocale <= numLocales; curLocale++)
     {
-        const char *name = plLocalization::GetLanguageName((plLocalization::Language)curLocale);
-        wchar_t *wName = hsStringToWString(name);
-        retVal.push_back(ST::string::from_wchar(wName));
-        delete [] wName;
+        retVal.push_back(plLocalization::GetLanguageName((plLocalization::Language)curLocale));
```

Comment:
```suggestion
        retVal.emplace_back(plLocalization::GetLanguageName((plLocalization::Language)curLocale));
```

Reply:
I'm pretty sure that won't make a difference in this case. Now that the `GetLanguageName` call is inline instead of stored in a variable, `push_back` can do a move instead of a copy. Using `emplace_back` here would just call the move constructor of `ST::string`, which does the same thing.

Reply:
Technically speaking, `push_back` means "copy." In practice, MSVC's `push_back` simply calls `emplace_back`, so there is no difference. However, I like to see `emplace_back` used to indicate "move/construct in place" and `push_back` for "copy".

Reply:
It's not just an MSVC thing - [`push_back(T&&)`](https://en.cppreference.com/w/cpp/container/vector/push_back) is part of C++11. But if you're using `emplace_back` as a convention here I have no problem with that.